backend/app/vending/__init__.py

```diff
@@ -191,6 +248,144 @@ def get_dashboard_link(login=Depends(login_state)) -> VendingRedirect:
+    except ValueError as val_err:
+        raise VendingError(error="bad-app-share") from val_err
+
+    (
+        fee_fixed_cost,
+        fee_cost_percent,
+        fee_prefer_percent,
+    ) = prices.flathub_fee_parameters(vend.currency)
+
+    return VendingDescriptor(
+        status="ok",
+        currency=vend.currency,
+        components=shares,
+        fee_fixed_cost=fee_fixed_cost,
+        fee_cost_percent=fee_cost_percent,
+        fee_prefer_percent=fee_prefer_percent,
+    )
+
+
+@router.post("app/{appid}/setup")
+def post_app_vending_setup(
+    appid: str, setup: VendingSetup, login=Depends(login_state)
+) -> VendingDescriptor:
+    """
+    Create/update the vending status for a given application.  Returns an error
+    if the appid is not known, or if it's already set up for vending with a
+    user other than the one calling this API.
+
+    If you do not have the right to set the vending status for this application
+    then you will also be refused.
+
+    In addition, if any of the currency or amount values constraints are violated
+    then you will get an error
+    """
+
+    if not login["state"].logged_in():
+        raise VendingError(error="not-logged-in")
+
+    vend = ApplicationVendingConfig.by_appid(db, appid)
+    if not vend:
+        # TODO: Check that the calling user owns the given appid
+        # TODO: Verify that the calling user is onboarded with stripe
+        vend = ApplicationVendingConfig(
+            user=login["user"].id,
+            appid=appid,
+            currency=setup.currency,
+            appshare=setup.appshare,
+            recommended_donation=setup.recommended_donation,
+            minimum_payment=setup.minimum_payment,
+        )
+    else:
+        if vend.user != login["user"]:
+            raise VendingError(error="user-mismatch")
+        vend.currency = setup.currency
+        vend.appshare = setup.appshare
+        vend.recommended_donation = setup.recommended_donation
+        vend.minimum_payment = setup.minimum_payment
+    try:
+        db.session.add(vend)
+        db.session.commit()
+    except Exception as base_exc:
+        raise VendingError(error="bad-values") from base_exc
+
+    return get_app_vending_status(appid)
+
+
+@router.get("app/{appid}/{currency}/{value}")
+def get_app_vending_split(appid: str, currency: str, value: int) -> VendingSplit:
+    """
+    Retrieve the actual split which would be invoked if the user specified that
+    they wished to spend the given amount on the given appid.  Note this is not
+    the exact shape that an invoice would have since donation vs. purchase may
+    need to be taken into account.
+    """
+    vend = ApplicationVendingConfig.by_appid(db, appid)
+    if not vend:
+        raise VendingError(error="not-found")
+    if vend.currency != currency:
+        raise VendingError(error="bad-currency")
+
+    try:
+        shares = prices.compute_app_shares(value, currency, appid, vend.appshare)
```

Comment:
Yes, I left it open which of the two of those you prefer.  However there is `GET /vendingapp/{appid}/{currency}/{value}` which returns the computed split if you want, for now, to prefer correctness over performance.

Reply:
If we're happy with multiple sources of truth I'll go with running the equivalent calculation client-side to be able to give live feedback without creating excess network traffic or having to do some form of debouncing (once using React 18 the new transitions feature might be an elegant way to handle updating a preview using requests to the backend).

Reply:
Having began implementing this client-side there are more configured variables involved (data from platforms.json, the flathub fixed cost and flathub share, etc.) on the backend than I anticipated. 

I think those will make maintaining parity more of an annoyance than expected. For now I'll extend this endpoint to also accept an appshare value (instead of being hardcoded to the one that's already set) and deal with the whole debouncing thing.

An alternative more involved solution is to expose all the needed variables from the backend such that they can be fetched statically and passed down to the client-side equivalent calculation.

Reply:
Whatever your preference is here, we can do - we can discuss at the meeting or offline.